packages/react-native-eudiw-pid/src/lib/base64.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * The code was extracted from:
+ * https://github.com/davidchambers/Base64.js
+ */
+
+const chars =
+  "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
+
+class InvalidCharacterError extends Error {
+  constructor(...args: string[]) {
+    super(...args);
+    this.name = "InvalidCharacterError";
+  }
+}
+
+function atob(input: string) {
+  const str = input.replace(/=+$/, "");
+  if (str.length % 4 === 1) {
+    throw new InvalidCharacterError(
+      "'atob' failed: The string to be decoded is not correctly encoded."
+    );
+  }
+  let output = ""; // eslint-disable-line functional/no-let
+  for (
+    // initialize result and counters
+    let bc = 0, bs = -1, buffer, idx = 0; // eslint-disable-line functional/no-let
+    // get next character
+    (buffer = str.charAt(idx++));
+    // character found in table? initialize bit storage and add its ascii value;
+    ~buffer && // eslint-disable-line no-bitwise
+    ((bs = bc % 4 ? bs * 64 + buffer : buffer),
+    // and if not first of each 4 characters,
+    // convert the first 8 bits to one ascii character
+    bc++ % 4)
+      ? (output += String.fromCharCode(255 & (bs >> ((-2 * bc) & 6)))) // eslint-disable-line no-bitwise
+      : 0
+  ) {
+    // try to find character in table (0-63, not found => -1)
+    buffer = chars.indexOf(buffer);
+  }
+  return output;
+}
```

Comment:
I can add the dependency; still, I cannot compare the implementation. `react-native-base64` is a [plain-javascript implementation](https://github.com/eranbo/react-native-base64) as well, and I see no big improvement in terms of reliability and performance.

The actual code is copy-pasted from https://github.com/auth0/jwt-decode. I can do the same here - without the need for a direct dependency (they are both MIT licensed).

Reply:
I think so! If it's defined on the browser then it's also on the RN. 
Maybe @hevelius @balanza know more!

Reply:
Maybe yes: https://solana.stackexchange.com/questions/6242/error-while-sending-transaction-with-solanaweb3-referenceerror-property-buff

Reply:
https://github.com/pagopa/io-app/blob/14e4af3430d07c33d0abbb5fd885d590e2138436/ts/features/fci/utils/signature.ts#L40

Reply:
Buffer is used in io-app via rn-nodeify https://github.com/pagopa/io-app/blob/14e4af3430d07c33d0abbb5fd885d590e2138436/package.json#L41 because not available by default in rn.

If we don't want to add an external dependency only way is to get our methods to do encoding/decoding of base64 string.